experimental/sdk/webpubsub-socketio-extension/src/SIO/index.ts

```diff
@@ -19,19 +27,52 @@ export async function useAzureSocketIOChain(
 ): Promise<SIO.Server> {
   debug(`useAzureSocketIOChain, webPubSubOptions: ${JSON.stringify(webPubSubOptions)}`);
   const engine = new WebPubSubEioServer(this.engine.opts, webPubSubOptions);
-  engine.attach(this["httpServer"], this["opts"]);
+  const httpServer = this["httpServer"];
+  engine.attach(httpServer, this["opts"]);
 
   // Using Tunnel
   if (engine.webPubSubConnectionManager.service instanceof InprocessServerProxy) {
     await engine.setup();
   }
 
+  // Add negotiate handler
+  debug("add negotiate handler");
+  const path = this._opts.path || "/socket.io";
+  const negotiatePathPrefix = path + (path.endsWith("/") ? "" : "/") + "negotiate/";
+
+  // current listeners = EIO handleRequest listeners (e.g. /socket.io) + other listeners from user
+  const listeners = httpServer.listeners("request").slice(0);
+  httpServer.removeAllListeners("request");
+  httpServer.on("request", (req: IncomingMessage, res: ServerResponse) => {
+    // negotiate handler
+    if (
+      webPubSubOptions.getGenerateClientTokenOptions &&
```

Comment:
Do you mean, even if user doesn't provide a option, we still handle `/negotiate` request and return a default endpoint with access token?
If so, I prefer not. No option means user doesn't want to use negotiate feature. Even he doesn't realize we put an extra `/negotiate` handler to his HTTP server. It could cause conflict potentially.